metrics-operator/controllers/common/providers/datadog/datadog.go

```diff
@@ -32,63 +39,119 @@ func (d *KeptnDataDogProvider) EvaluateQuery(ctx context.Context, metric metrics
 		return "", nil, err
 	}
 	qURL := provider.Spec.TargetServer + "/api/v1/query?from=" + strconv.Itoa(int(fromTime)) + "&to=" + strconv.Itoa(int(toTime)) + "&query=" + url.QueryEscape(metric.Spec.Query)
-	req, err := http.NewRequestWithContext(ctx, "GET", qURL, nil)
-	if err != nil {
-		d.Log.Error(err, "Error while creating request")
-		return "", nil, err
-	}
-
 	apiKeyVal, appKeyVal, err := getDDSecret(ctx, provider, d.K8sClient)
 	if err != nil {
 		return "", nil, err
 	}
 
-	req.Header.Set("Accept", "application/json")
-	req.Header.Set("Dd-Api-Key", apiKeyVal)
-	req.Header.Set("Dd-Application-Key", appKeyVal)
-
-	res, err := d.HttpClient.Do(req)
+	b, err := d.executeQuery(ctx, qURL, apiKeyVal, appKeyVal)
 	if err != nil {
-		d.Log.Error(err, "Error while creating request")
 		return "", nil, err
 	}
-	defer func() {
-		err := res.Body.Close()
-		if err != nil {
-			d.Log.Error(err, "Could not close request body")
-		}
-	}()
 
-	b, _ := io.ReadAll(res.Body)
 	result := datadogV1.MetricsQueryResponse{}
 	err = json.Unmarshal(b, &result)
 	if err != nil {
-		d.Log.Error(err, "Error while parsing response")
+		d.Log.Error(err, errParseRes)
```

Comment:
I believe we can remove the `d.Log.Error/Info` statements in the error cases as any errors returned by `EvaluateQuery` will be logged by the metrics controller anyway